account_asset_management/models/account_asset.py

```diff
@@ -0,0 +1,1103 @@
+        comodel_name='account.asset',
+        string='Parent Asset', readonly=True,
+        states={'draft': [('readonly', False)]},
+        domain=[('type', '=', 'view')],
+        ondelete='restrict',
+        index=True,
+    )
+    parent_left = fields.Integer(index=True)
+    parent_right = fields.Integer(index=True)
+    child_ids = fields.One2many(
+        comodel_name='account.asset',
+        inverse_name='parent_id',
+        string='Child Assets')
+    date_start = fields.Date(
+        string='Asset Start Date', readonly=True,
+        states={'draft': [('readonly', False)]},
+        help="You should manually add depreciation lines "
+             "with the depreciations of previous fiscal years "
+             "if the Depreciation Start Date is different from the date "
+             "for which accounting entries need to be generated.")
+    date_remove = fields.Date(string='Asset Removal Date', readonly=True)
+    state = fields.Selection(
+        selection=[
+            ('draft', 'Draft'),
+            ('open', 'Running'),
+            ('close', 'Close'),
+            ('removed', 'Removed'),
+        ], string='Status', required=True, default='draft', copy='draft',
+        help="When an asset is created, the status is 'Draft'.\n"
+             "If the asset is confirmed, the status goes in 'Running' "
+             "and the depreciation lines can be posted "
+             "to the accounting.\n"
+             "If the last depreciation line is posted, "
+             "the asset goes into the 'Close' status.\n"
+             "When the removal entries are generated, "
+             "the asset goes into the 'Removed' status.")
+    active = fields.Boolean(default=True)
+    partner_id = fields.Many2one(
+        comodel_name='res.partner', string='Partner', readonly=True,
+        states={'draft': [('readonly', False)]})
+    method = fields.Selection(
+        selection=lambda self: self.env[
+            'account.asset.profile']._selection_method(),
+        string='Computation Method',
+        required=True, readonly=True,
+        states={'draft': [('readonly', False)]}, default='linear',
+        help="Choose the method to use to compute "
+             "the amount of depreciation lines.\n"
+             "  * Linear: Calculated on basis of: "
+             "Gross Value / Number of Depreciations\n"
+             "  * Degressive: Calculated on basis of: "
+             "Residual Value * Degressive Factor"
+             "  * Degressive-Linear (only for Time Method = Year): "
+             "Degressive becomes linear when the annual linear "
+             "depreciation exceeds the annual degressive depreciation")
+    method_number = fields.Integer(
+        string='Number of Years', readonly=True,
+        states={'draft': [('readonly', False)]}, default=5,
+        help="The number of years needed to depreciate your asset")
+    method_period = fields.Selection(
+        selection=lambda self: self.env[
```

Comment:
Do we really need to pass a `lambda` here? I see that the results of `_selection_method_period` are hardcoded, wouldn't it be better to define the function before the field and just pass it here?